### PR TITLE
pdf version specs

### PR DESCRIPTION
**Pull request by Arshitha**
This is a PR almost identical to PR [#400 updated-pdf-version-specs](https://github.com/bids-standard/bids-specification/pull/400) except that this has much cleaner commit history and was created to address the issue of unwanted commits in the above referenced PR. 

To gain a better understanding of why certain decisions were made while fixing the issue of generating pdf version of the specs, discussions in the following issues would be useful references: 
- pdf version of spec bids-standard/bids-specification#135
- updated-pdf-version-specs bids-standard/bids-specification#400

### Bugs Fixed: 
- Stable Table of Contents page
- Cover page 
- Non-tabular text overflow 
- Overlapping columns of tables (owing to small widths of the pipe tables in the markdown files)
  - Markdown files were edited (one-time edit) to increase the column widths of pipe tables
  - Does not affect the read the docs version of specs. However, if any tables were added to the specs in future, there needs to a standard that works for both pdf AND read the docs version of the specs. 
- Automating header to include the latest release version number and build date (pulled from `mkdocs.yml`)

### Enhancements required (non-exhaustive list): 
- Internal links referencing either the same markdown file or other markdown files break during conversion from markdown to pdf. Currently, all of the internal links are removed and replaced by the accompanying text. 
- Superscripts are lost in conversion as well 
- Tried a couple different family of fonts to fix the missing character warnings for emojis used in the markdowns with no success. It would be ideal to fix this to match the Read The Docs version more closely
- Entity table in Appendix IV has overlapping unreadable columns since it's markdown equivalent has more than 10 columns. Refer to issue for more discussion

### CODE REVIEW

**Comment by Arshitha** [comment]
The rendered pdf can be found under the `artifacts` tab of the circleci `build_docs_pdf` test details. 
For easy reference: [bids-spec.pdf]( https://1927-150465237-gh.circle-artifacts.com/0/bids-spec.pdf)

**Comment by sappelhoff** [comment]
@effigies @yarikoptic @nicholst @franklin-feingold @robertoostenveld 

I am pinging you because you have been involved in discussions about the PDF generation in the past (please help me to ping those people I forgot).

We have gone through several iterations with this PR now with @Arshitha busily fixing bugs and shaping the PR to our requests.

There is still a lot to be done (as outlined in the OP above by @Arshitha ) ... but I think we should merge this PR first, because it provides a solid foundation.

If you think so too, please review and approve ... disapprove only if you find issues that should definitely not make it into the bids-specification repo ... and then tell us about these issues so we can stay active :-)

---

PS: I tried rebasing this PR and it was hell (I failed for now) ... if somebody else wants to train some git kung fu, please try to rebase this PR and make a new PR that we can merge.

Else we can look into making a merge commit --> but I already see several things that'd need to be resolved (see my code comments).

File: .circleci/config.yml [hunk 1]
```diff
@@ -57,7 +73,7 @@ jobs:
            working_directory: ~/build
            command: |
              if (git log -1 --pretty=%s | grep Merge*) && (! git log -1 --pretty=%b | grep REL:) ; then
-             github_changelog_generator --user bids-standard --project bids-specification --token ${CHANGE_TOKEN} --output ~/build/CHANGES.md --base ~/build/src/pregh-changes.md --header-label Changelog --no-issues --no-issues-wo-labels --no-filter-by-milestone --no-compare-link --pr-label ""
+             github_changelog_generator --user bids-standard --project bids-specification --token ${CHANGE_TOKEN} --output ~/build/CHANGES.md --base ~/build/src/pregh-changes.md --header-label "# Changelog" --no-issues --no-issues-wo-labels --no-filter-by-milestone --no-compare-link --pr-label ""
```
Comment (sappelhoff) [inline .circleci/config.yml]: I am wondering why this line is highlighted as a diff.

In `master`, we should have the exact same line: https://github.com/bids-standard/bids-specification/blob/6bd2ad596f96706c2be4db8cd6551cf85418edd8/.circleci/config.yml#L60

... so there shouldn't be a change, actually :thinking:

**Comment by Arshitha** [comment]
>  PS: I tried rebasing this PR and it was hell (I failed for now) ... if somebody else wants to train some git kung fu, please try to rebase this PR and make a new PR that we can merge.
> 
> Else we can look into making a merge commit --> but I already see several things that'd need to be resolved (see https://github.com/bids-standard/bids-specification/pull/427/files#r384366609).

@sappelhoff would it be easier if I were to create a new branch from master and then just add all of the new/changed files to this new branch to create a PR from there? That way, it won't have more than 2-3 commits and therefore, easier to rebase/merge? 

This PR can then be deleted?

File: .circleci/config.yml (hunk 1 above)
Comment (Arshitha) [reply .circleci/config.yml]: I am not sure why that's the case. I remember having synced my fork with the latest commits on master, but perhaps I missed something when cherry picking commits from here https://github.com/bids-standard/bids-specification/pull/400 ?

**Comment by Arshitha** [comment]
Also, most of the conflicts to be resolved, if this were to be merged, would be column width adjustments in the markdown files.

**Comment by sappelhoff** [comment]
> @sappelhoff would it be easier if I were to create a new branch from master and then just add all of the new/changed files to this new branch to create a PR from there? 

@Arshitha that's how I would do it ... :-) but we could perhaps also just make a merge commit without rebasing this PR. I am only puzzled by the `"# Changelog"` diff, which shouldn't be a diff :thinking: 

perhaps @effigies can comment, as he has more git experience.

**Comment by Arshitha** [comment]
@sappelhoff can we go ahead and merge by creating a new PR with only a couple of commits or do we still try rebasing or merging this one?

**Comment by sappelhoff** [comment]
@Arshitha I did not resolve the question with the unexpected diff that's showing up (https://github.com/bids-standard/bids-specification/pull/427/files#r384366609).

So yes, perhaps it'd be best if you could 
1. get your fork up to date with current master
1. make a new branch
1. add the new contents of the PDF generation
1. make a new PR

I'll be sure to make a quick check then and merge. We have all the reviews (and approvals) we need :-)

**Pull request closed without merging**